File: 2563/Solution.java
class Solution {
    public long countFairPairs(int[] nums, int lower, int upper) {
        Arrays.sort(nums);
        return find(nums, upper) - find(nums, lower-1);
    }
    
    long find(int[] nums, int upper){
        long res = 0;
        int i = 0, j = nums.length-1;
        while(i < j){
            while(i < j && nums[i] + nums[j] > upper)
                j--;
                
            res += j - i;
            i++;
        }
        return res;
    }
}

File: 2563/Solution.go
func countFairPairs(nums []int, lower int, upper int) int64 {
    sort.Ints(nums)
    return find(nums, upper) - find(nums, lower - 1)
}

func find(nums []int, target int) int64 {
    var res int64
    for i, j := 0, len(nums) - 1; i < j; {
        if nums[i] + nums[j] > target {
            j--
        } else {
            res += int64(j - i)
            i++
        }
    }
    return res
}
